io.c

```diff
@@ -8828,155 +8879,28 @@ rb_io_make_open_file(VALUE obj)
 
 /*
  *  call-seq:
- *     IO.new(fd [, mode] [, opt])   -> io
- *
- *  Returns a new IO object (a stream) for the given integer file descriptor
- *  +fd+ and +mode+ string.  +opt+ may be used to specify parts of +mode+ in a
- *  more readable fashion.  See also IO.sysopen and IO.for_fd.
- *
- *  IO.new is called by various File and IO opening methods such as IO::open,
- *  Kernel#open, and File::open.
- *
- *  === Open Mode
+ *    IO.new(fd, mode = 'r', **opts) -> io
  *
- *  When +mode+ is an integer it must be combination of the modes defined in
- *  File::Constants (+File::RDONLY+, <code>File::WRONLY|File::CREAT</code>).
- *  See the open(2) man page for more information.
+ *  Returns a new \IO object (a stream) for the given arguments.
  *
- *  When +mode+ is a string it must be in one of the following forms:
+ *  Argument +fd+ must be a valid file descriptor (integer):
```

Comment:
Here, again, we have a problem of "why": creating an `IO` from raw file descriptor is rarely needed in a more advanced code, but we leaving reader with either the question "why is it necessary", or, worse, with the impression that opening a file in Ruby is always as cumbersome as `syscall` + `IO.new`. I suggest adjusting the intro to something to the meaning of...

> Creates a new IO object from a file descriptor given by a system call. 

...and maybe at the end of the docs:

> Note that most of the time it is easier to use more high-level methods like File::open for creating IO streams. ::new is useful for interaction with low-level libraries.

Or something like that.

Reply:
Modified.